lib/json_api_controller/precondition_check.rb

```diff
@@ -0,0 +1,28 @@
+module JsonApiController
+  module PreconditionCheck
+    def precondition_check
+      if !precondition
+        raise PreconditionNotPresent, precondition_error_msg
+      elsif precondition_fails?
+        raise PreconditionFailed
+      end
+    end
+    
+    def precondition
+      case action_name
+      when "show"
+        request.headers["If-Modified-Since"]
+      else
+        request.headers["If-Unmodified-Since"]
```

Comment:
Can you update the other issue so @brian-c knows which header to use (in case he doesn't read the apiary docs).